src/sources/http_client/client.rs

```diff
@@ -331,10 +331,12 @@ impl http_client::HttpClientContext for HttpClientContext {
         let body = String::from_utf8_lossy(body);
         buf.extend_from_slice(body.as_bytes());
 
-        // decode and enrich
-        let mut events = self.decode_events(&mut buf);
-        self.enrich_events(&mut events);
+        let events = self.decode_events(&mut buf);
 
         Some(events)
     }
+
+    fn enrich<'a>(&'a mut self, events: &'a mut Vec<Event>) {
+        self.enrich_events(events);
+    }
```

Comment:
I'm not seing an advantage to having the separate `enrich_events` function anymore. I'd originally put `decode_events` and `enrich_events` in the impl so as to breakup the `on_response`.

But since there is now an `enrich` trait function implementation splitting that up, this extra hop only serves to obfuscate imo.